helm/cluster/templates/clusterapi/controlplane/kamajicontrolplane.yaml

```diff
@@ -0,0 +1,141 @@
+{{- if and
+  $.Values.providerIntegration.resourcesApi.controlPlaneResource.enabled
+  (eq $.Values.providerIntegration.resourcesApi.controlPlaneResource.provider "kamaji")
+}}
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha1
+kind: KamajiControlPlane
```

Comment:
i have based this off of https://github.com/clastix/cluster-api-kamaji-vsphere/blob/master/charts/capi-kamaji-vsphere/templates/kamaji-control-plane.yaml since we will work with the vsphere provider first